### DISCUSSION TITLE
Support for EA games

### DISCUSSION THREAD

**Discussion by ahmouse15** [Heroic-Games-Launcher/HeroicGamesLauncher#3867] [Ideas]
Hello,

Recently I began working on [yaro](https://github.com/ahmouse15/yaro), a command-line launcher for EA games. It's still _very_ early stages, but I thought I'd mention it here since many of you have experience with reverse engineering + game launchers, plus I'd like to see Heroic integrate with it once it's useable. With that said, if anyone has insight into EA's APIs, or any other knowledge of Origin/EA App, I would love to hear from you.

It's been in the works for about a month (mostly analysis rather than programming), and have reached the stage of logging in. The next major focus will be on understanding & implementing game downloads.

**Comment by Lariaa** [comment]
we are working with https://twitter.com/maximalauncher from the https://uplink.kyber.gg/news/features-overview/ crew to enable this in the future.

**Reply by ahmouse15** [reply]
Very cool! Any idea when that launcher itself will be available? And are there any similar projects for Ubisoft Connect?

**Reply by Lariaa** [reply]
No i don't. There are things like left to be done like figuring out file formats like .eapatchdiff or the ea installer.

For Ubi we have https://github.com/UplayDB/ & https://github.com/YoobieRE/ , which are currently stuck on intercepting and decoding the connection between game and launcher via NamedPipes so launching becomes possible.

Help is always [welcome](https://github.com/Lariaa/GameLauncherResearch/wiki)

**Reply by UniK** [reply]
> we are working with https://twitter.com/maximalauncher from the https://uplink.kyber.gg/news/features-overview/ crew to enable this in the future.

How is the collaboration progressing?

Based on my research, it appears to have stalled before it even began. The source code repository no longer exists. 
Here’s the link: `https://github.com/armchairdevelopers/maxima`

**Reply by ahmouse15** [reply]
Have there been any updates? It would be nice if it was open source so others could contribute! I went as far as extracting the secret key used to authenticate users into the EA app, allowing me to generate a login session. However, I was unable to decrypt the TLS session for downloading games, as the usual proxy methods don't seem to work
